File: src/com/eagle/programmar/Python/Statements/Python_WithStatement.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Dec 8, 2013

package com.eagle.programmar.Python.Statements;

import com.eagle.programmar.Python.Python_Expression;
import com.eagle.programmar.Python.Python_Statement;
import com.eagle.programmar.Python.Terminals.Python_EndOfLine;
import com.eagle.programmar.Python.Terminals.Python_Keyword;
import com.eagle.tokens.SeparatedList;
import com.eagle.tokens.TokenList;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationColon;
import com.eagle.tokens.punctuation.PunctuationComma;

public class Python_WithStatement extends TokenSequence
{
	public @S(10) @OPT Python_Keyword ASYNC = new Python_Keyword("async");
	public @S(20) @DOC("compound_stmts.html#the-with-statement") Python_Keyword WITH = new Python_Keyword("with");
	public @S(30) SeparatedList<Python_WithItem,PunctuationComma> withItems;
	public @S(40) PunctuationColon colon;
	public @S(50) Python_EndOfLine eoln;
	public @S(60) TokenList<Python_Statement> statements;
	
	public static class Python_WithItem extends TokenSequence
	{
		public @S(10) Python_Expression condition;
		public @S(20) @OPT Python_WithItemAs withItemAs;
		
		public static class Python_WithItemAs extends TokenSequence
		{
			public @S(10) Python_Keyword AS = new Python_Keyword("as");
			public @S(20) Python_Expression expression;
		}
	}
}

File: csrc/com/eagle/programmar/Python/Statements/Python_WithStatement.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Dec 8, 2013

namespace com.eagle.programmar.Python.Statements
{
	using Python_Expression = com.eagle.programmar.Python.Python_Expression;
	using Python_Keyword = com.eagle.programmar.Python.Terminals.Python_Keyword;
	using SeparatedList = com.eagle.tokens.SeparatedList;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationColon = com.eagle.tokens.punctuation.PunctuationColon;
	using PunctuationComma = com.eagle.tokens.punctuation.PunctuationComma;

	public class Python_WithStatement : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) @OPT Python_Keyword ASYNC = new com.eagle.programmar.Python.Terminals.Python_Keyword("async");
		public  OPT;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @DOC("compound_stmts.html#the-with-statement") com.eagle.programmar.Python.Terminals.Python_Keyword WITH = new com.eagle.programmar.Python.Terminals.Python_Keyword("with");
		public @DOC("compound_stmts.html#the-with-statement") Python_Keyword WITH = new Python_Keyword("with");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.tokens.SeparatedList<Python_WithItem, com.eagle.tokens.punctuation.PunctuationComma> withItems;
		public SeparatedList<Python_WithItem, PunctuationComma> withItems;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.tokens.punctuation.PunctuationColon colon;
		public PunctuationColon colon;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) Python_StatementBlock statements;
		public Python_StatementBlock statements;

		public static class Python_WithItem extends TokenSequence
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.Python.Python_Expression condition;
			public Python_Expression condition;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) @OPT Python_WithItemAs withItemAs;
			public @OPT Python_WithItemAs withItemAs;

			public static class Python_WithItemAs extends TokenSequence
			{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.Python.Terminals.Python_Keyword AS = new com.eagle.programmar.Python.Terminals.Python_Keyword("as");
				public Python_Keyword AS = new Python_Keyword("as");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.programmar.Python.Python_Expression expression;
				public Python_Expression expression;
			}
		}
	}

}
